### PR TITLE
Add doc for CA2260

### PR DESCRIPTION
**Pull request by buyaa-n**
## Summary

Add doc for new rule CA2260

Related to https://github.com/dotnet/runtime/issues/69775

The analyzer PR: https://github.com/dotnet/roslyn-analyzers/pull/6126

### CODE REVIEW

File: docs/fundamentals/code-analysis/quality-rules/ca2260.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+|                                     | Value                                |
+| ----------------------------------- | ------------------------------------ |
+| **Rule ID**                         | CA2260                               |
+| **Category**                        | [Usage](usage-warnings.md)           |
+| **Fix is breaking or non-breaking** | Non-breaking                         |
+
+## Cause
+
+When not passing the type itself as a type parameter when implementing a generic math interface that requires self recurring type parameter.
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/ca2260.md:24]:
```suggestion
This rule fires when you implement a generic math interface that requires a self-recurring type parameter and you don't pass the type itself as the type parameter.
```

File: docs/fundamentals/code-analysis/quality-rules/ca2260.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+using System;
+
+// Warns: The 'IParsable<TSelf>' requires the 'TSelf' type parameter to be filled with the derived type 'MyDate'
+public readonly struct MyDate : IParsable<DateOnly>
+{ ... }
+```
+
+**Fix**:
+
+If your array's elements are larger than one byte in size, you can multiply the length of the array by the element size to get the number of bytes.
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/ca2260.md:48]: ??

File: docs/fundamentals/code-analysis/quality-rules/index.md [hunk 3]
```diff
@@ -203,6 +203,7 @@ The following table lists code quality analysis rules.
...
 > | [CA2257: Members defined on an interface with 'DynamicInterfaceCastableImplementationAttribute' should be 'static'](ca2257.md) | Since a type that implements `IDynamicInterfaceCastable` may not implement a dynamic interface in metadata, calls to an instance interface member that is not an explicit implementation defined on this type are likely to fail at run time. Mark new interface members `static` to avoid run-time errors. |
 > | [CA2258: Providing a 'DynamicInterfaceCastableImplementation' interface in Visual Basic is unsupported](ca2258.md) | Providing a functional `DynamicInterfaceCastableImplementationAttribute`-attributed interface requires the Default Interface Members feature, which is unsupported in Visual Basic. |
+> | [CA2260: Implement Generic Math Interfaces Correctly](ca2260.md) | Generic math interfaces require the derived type itself to be used for the self recurring type parameter |
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/index.md:206]:
```suggestion
> | [CA2260: Implement generic math interfaces correctly](ca2260.md) | Generic math interfaces require the derived type itself to be used for the self-recurring type parameter. |
```

File: docs/fundamentals/code-analysis/quality-rules/usage-warnings.md [hunk 4]
```diff
@@ -61,3 +61,4 @@ Usage rules support proper usage of .NET.
...
 |[CA2257: Members defined on an interface with 'DynamicInterfaceCastableImplementationAttribute' should be 'static'](ca2257.md) | Since a type that implements `IDynamicInterfaceCastable` may not implement a dynamic interface in metadata, calls to an instance interface member that is not an explicit implementation defined on this type are likely to fail at run time. Mark new interface members `static` to avoid run-time errors. |
 |[CA2258: Providing a 'DynamicInterfaceCastableImplementation' interface in Visual Basic is unsupported](ca2258.md) | Providing a functional `DynamicInterfaceCastableImplementationAttribute`-attributed interface requires the Default Interface Members feature, which is unsupported in Visual Basic. |
+|[CA2260: Implement Generic Math Interfaces Correctly](ca2260.md) | Generic math interfaces require the derived type itself to be used for the self recurring type parameter |
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/usage-warnings.md:64]:
```suggestion
| [CA2260: Implement generic math interfaces correctly](ca2260.md) | Generic math interfaces require the derived type itself to be used for the self-recurring type parameter. |
```

File: docs/fundamentals/code-analysis/quality-rules/ca2260.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+| **Category**                        | [Usage](usage-warnings.md)           |
+| **Fix is breaking or non-breaking** | Non-breaking                         |
+
+## Cause
+
+When not passing the type itself as a type parameter when implementing a generic math interface that requires self recurring type parameter.
+
+## Rule description
+
+Some generic math interfaces introduce static abstract members, the only way to access those static members is through a generic constraint, the generic constraints implements Curiously Recurring Template Pattern (CRTP), therefore require the derived type itself to be used for the self recurring type parameter. If a type implements such interface without passing required type parameter and CA2260 is ignored the code would compile successfully but the static abstract will not me accessible therefore the type will not be usable, compiler would warn with CS0315 on such usage.
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/ca2260.md:28]:
```suggestion
Some generic math interfaces introduce static abstract members. The only way to access those static members is through a generic constraint that implements the "curiously recurring template pattern" (CRTP). Therefore, the derived type itself must be used for the self-recurring type parameter. If a type implements such an interface without passing the required type parameter and CA2260 is ignored, the code will compile successfully but the static abstract will not be accessible. Thus, the type will not be usable. The compiler emits a warning with ID CS0315 on such usage.
```

**Review by gewarren — COMMENTED** [review]
Thanks @buyaa-n - I left some grammatical suggestions.

File: docs/fundamentals/code-analysis/quality-rules/ca2260.md (hunk 2 above)
Comment (buyaa-n) [reply docs/fundamentals/code-analysis/quality-rules/ca2260.md:48]: Left over from copy paste 😛 
```suggestion
Pass the `MyDate` type as type parameter for the `IParsable<TSelf>` interface
```

File: docs/fundamentals/code-analysis/quality-rules/ca2260.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+ms.topic: reference
+f1_keywords:
+- CA2260
+- ImplementGenericMathInterfacesCorrectly
+helpviewer_keywords:
+- ImplementGenericMathInterfacesCorrectly
+- CA2260
+author: buyaa-n
+---
+# CA2260: Implement Generic Math Interfaces Correctly
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/ca2260.md:14]:
```suggestion
# CA2260: Implement generic math interfaces correctly
```

File: docs/fundamentals/code-analysis/quality-rules/ca2260.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
+---
+title: "CA2260: Implement Generic Math Interfaces Correctly"
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/ca2260.md:2]:
```suggestion
title: "CA2260: Implement generic math interfaces correctly"
```

File: docs/fundamentals/code-analysis/quality-rules/ca2260.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+description: "Generic math interfaces require the derived type itself to be used for the self recurring type parameter"
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/ca2260.md:3]:
```suggestion
description: "Learn about code analysis rule CA2260 - Generic math interfaces require the derived type itself to be used for the self recurring type parameter."
```

File: docs/fundamentals/code-analysis/quality-rules/ca2260.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+using System;
+
+// Warns: The 'IParsable<TSelf>' requires the 'TSelf' type parameter to be filled with the derived type 'MyDate'
+public readonly struct MyDate : IParsable<DateOnly>
+{ ... }
+```
+
+**Fix**:
+
+Pass the `MyDate` type as type parameter for the `IParsable<TSelf>` interface
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/ca2260.md:48]:
```suggestion
Pass the `MyDate` type as the type parameter for the `IParsable<TSelf>` interface.
```

**Review by gewarren — APPROVED** [review]

**Pull request merged**